contracts/SignerBondsUniswapV2.sol

```diff
@@ -0,0 +1,260 @@
+    // Determines the deadline in which the swap transaction has to be mined.
+    // If that deadline is exceeded, transaction will be reverted.
+    uint256 public swapDeadline = 20 minutes;
+
+    event UniswapV2SwapExecuted(uint256[] amounts);
+
+    constructor(IUniswapV2Router _uniswapRouter, CoveragePool _coveragePool) {
+        uniswapRouter = _uniswapRouter;
+        targetTokenRecipient = address(_coveragePool.assetPool());
+        targetToken = address(_coveragePool.collateralToken());
+        uniswapPair = IUniswapV2Pair(
+            computePairAddress(
+                _uniswapRouter.factory(),
+                _uniswapRouter.WETH(),
+                targetToken
+            )
+        );
+    }
+
+    /// @notice Swaps signer bonds.
+    /// @dev Adds incoming bonds to the overall contract balance.
+    function swapSignerBonds() external payable override {}
+
+    /// @notice Sets the maximum price impact allowed for a swap transaction.
+    /// @param _maxAllowedPriceImpact Maximum allowed price impact specified
+    ///        in basis points. Value of this parameter must be between
+    ///        0 and 10000 (inclusive). It should be chosen carefully as
+    ///        high limit level will accept transactions with high volumes.
+    ///        Those transactions may result in poor execution prices. Very low
+    ///        limit will force low swap volumes. Limit equal to 0 will
+    ///        effectively make swaps impossible.
+    function setMaxAllowedPriceImpact(uint256 _maxAllowedPriceImpact)
+        external
+        onlyOwner
+    {
+        require(
+            _maxAllowedPriceImpact <= BASIS_POINTS_DIVISOR,
+            "Maximum value is 10000 basis points"
+        );
+        maxAllowedPriceImpact = _maxAllowedPriceImpact;
+    }
+
+    /// @notice Sets the slippage tolerance for a swap transaction.
+    /// @param _slippageTolerance Slippage tolerance in basis points. Value of
+    ///        this parameter must be between 0 and 10000 (inclusive). It
+    ///        should be chosen carefully as transactions with high slippage
+    ///        tolerance result in poor execution prices. On the other hand,
+    ///        very low slippage tolerance may cause transactions to be
+    ///        reverted frequently. Slippage tolerance equal to 0 is possible
+    ///        and disallows any slippage to happen on the swap at the cost
+    ///        of higher revert risk.
+    function setSlippageTolerance(uint256 _slippageTolerance)
+        external
+        onlyOwner
+    {
+        require(
+            _slippageTolerance <= BASIS_POINTS_DIVISOR,
+            "Maximum value is 10000 basis points"
+        );
+        slippageTolerance = _slippageTolerance;
+    }
+
+    /// @notice Sets the deadline for a swap transaction.
+    /// @param _swapDeadline Swap deadline in seconds. Value of this parameter
+    ///        should be grater than 0. It should be chosen carefully as
```

Comment:
> Value of this parameter should be grater than 0

Do we want to put a check `require` for it?

Reply:
It should be greater but doesn't have to. If we allow `0` we can allow the government to actually block swaps for some time. If the government will want to resign from this privilege, we can add the `require`.

Reply:
Ok, can we maybe add this info, something like "should be greater equal than 0". By reading this sentence in a comment I though it's a strict `>`, not `>=`?

Reply:
Ok done.